tests/test_twitter_ads_bookmark.py

```diff
@@ -146,6 +146,31 @@ def test_run(self):
 
                         # Verify that all foreign keys in the 2nd sync are available in in 1st sync also. 
                         self.assertTrue(second_sync_child_foreign_keys.issubset(first_sync_child_foreign_keys))
+
+                        # Collect child key-properties from 1st sync records based on foreign keys of 2nd sync records
+                        first_sync_pks = set()
+                        for record in first_sync_messages:
+                            if record['line_item_id'] in second_sync_child_foreign_keys:
```

Comment:
I think this conditional should be checking against the parent stream records, not the child stream. You need a set of primary keys based off the messages that are sent to the target for the `line_items` stream. But only the messages with replication keys greater than or equal to the state you inject prior to the 2nd sync.

Reply:
We have already added an [assertion ](https://github.com/singer-io/tap-twitter-ads/blob/21b0ab8a07517dae50f511c16283cf8ac265473f/tests/test_twitter_ads_bookmark.py#L165-L173)to verify that the parent record's replication key value is greater than the child's bookmark value. In this assertion, we are validating the id of the parent that is available in the records of the child(2nd sync).